Custom Item Addition: Thomas Calloway OII Coffee Cup

CallowaySCG opened the pull request:
<!--
DO NOT REPLACE THE ACTUAL FIELDS
REPLACE THE ENTIRE LINES (INCLUDING ARROWS) THAT SAY REPLACE THIS
FAILURE TO DO SO THIS WILL RESULT IN INFINITE IN INEXORABLE DERISION 
-->

**Link to thread:**  
https://forums.baystation12.net/threads/oii-coffee-cup-thomas-calloway-thomas_calloway.8619/
**Approving maintainer**:  
Eckles
**I have tested this item:**:  
No, I am uncertain how to test the item at this time.

babydoll commented:
Note that mugs don't currently work properly in this system; see #207 and https://github.com/Baystation12/Baystation12/blob/dev/code/game/objects/items/weapons/lighter.dm#L193 & https://github.com/Baystation12/custom-items/pull/202

CallowaySCG commented:
Is there any way to get around that issue, or will it just be a paperweight? It would be neat to put on the desk, even if it does not work. 

Is it just coffee mugs, or is it all liquid containers? I could make a thermos or the like, potentially.

babydoll commented:
Fixing the issue probably wouldn't be too hard. Head over to #coding channel and ask for help, or I can probably give it a go sometime soon.

afterthought2 commented on config/custom_items/items/thomascalloway-oiicoffeecup.json:
```diff
@@ -0,0 +1,8 @@
+  "item_name" :       "OII coffee cup",
+  "item_desc" :       "This sleek, dark grey coffee cup is branded with the three-letter initialism of the Office of Interstellar Intelligence: OII.",
+  "item_icon_state" : "oii_coffee_cup",
+  "item_path" :       "/obj/item/weapon/reagent_containers/food/drinks/glass2/coffeecup/OII"
```
The correct type path is `/obj/item/weapon/reagent_containers/food/drinks/glass2/coffeecup`.

CallowaySCG replied:
You're right! I'm only marginally familiar with GitHub and the process--do I need to delete this pull request and resubmit or can you edit the line?

CallowaySCG replied:
I edited the line, this is flagged as outdated. The new path is exactly as you provided it, omitting /OII from the original.

babydoll commented:
Also the icon that goes in the `custom_items_mob.dmi` is the one that gets displayed in your character's hands. You probably don't want it to be a giant mug.

CallowaySCG commented:
Reasonable point! I'll edit that and resubmit as well.

afterthought2 commented:
Testing travis; ignore.

afterthought2 commented:
Please try going to the Travis website (https://travis-ci.com/Baystation12/custom-items should work) and clicking log in with github near the top right. After that, try to either force-push or close-reopen this PR.

CallowaySCG commented:
With Myazaki's help behind the scenes, I squashed my commits and removed the mob icon changes. The mob icon did not need to be included as coffee mugs are designed to be invisible in hands.
